apps/freenet-ping/web/src/main.rs

```diff
@@ -0,0 +1,151 @@
+use std::{collections::HashSet, time::Duration};
+
+use clap::Parser;
+use freenet_stdlib::{
+    client_api::{ClientRequest, ContractRequest, ContractResponse, HostResponse, WebApi},
+    prelude::{
+        ContractContainer, ContractKey, Parameters, RelatedContracts, StateDelta, UpdateData,
+        WrappedState,
+    },
+};
+use names::Generator;
+
+#[derive(Debug, Default, serde::Serialize, serde::Deserialize)]
+#[serde(transparent)]
+struct Ping {
+    from: HashSet<String>,
```

Comment:
Some improvements we can do:
-  store the time of the update (in UTC)
- when we update ensure that older than 1hour entries are dropped